src/instrument.js

```diff
@@ -322,21 +322,39 @@ export function liftReducerWith(reducer, initialCommittedState, monitorReducer,
         break;
       }
       case ActionTypes.IMPORT_STATE: {
-        // Completely replace everything.
-        ({
-          monitorState,
-          actionsById,
-          nextActionId,
-          stagedActionIds,
-          skippedActionIds,
-          committedState,
-          currentStateIndex,
-          computedStates
-        } = liftedAction.nextLiftedState);
-
-        if (liftedAction.noRecompute) {
-          minInvalidatedStateIndex = Infinity;
+        if (Array.isArray(liftedAction.nextLiftedState)) {
+          // recompute array of actions
+          actionsById = { 0: liftAction(INIT_ACTION) };
+          nextActionId = 1;
+          stagedActionIds = [0];
+          skippedActionIds = [];
+          currentStateIndex = 0;
+          computedStates = [];
+          minInvalidatedStateIndex = 0;
+          // iterate through actions
+          liftedAction.nextLiftedState.forEach(action => {
+            actionsById[nextActionId] = action;
```

Comment:
Should be `actionsById[nextActionId] = liftAction(action);`. I contains more information than just actions, [see the api](https://github.com/zalmoxisus/redux-devtools-instrument/tree/track-perf#api).